File: app/Models/Metier.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Metier extends Model
{
    use HasFactory;

    protected $table = 'metiers';

    protected $fillable = [
        'nom',
        'description',
        'salaire_min',
        'salaire_moyen',
        'salaire_max',
        'duree_estimee',
    ];

    public function competences(): BelongsToMany
    {
        return $this->belongsToMany(Competence::class);
    }

    public function parcoursEtudes(): BelongsToMany
    {
        return $this->belongsToMany(ParcoursEtude::class, 'metier_parcours_etude');
    }

    public function ecoles(): BelongsToMany
    {
        return $this->belongsToMany(Ecole::class);
    }

    public function roadmapEtapes(): HasMany
    {
        return $this->hasMany(RoadmapEtape::class)->orderBy('ordre');
    }
}

File: flutter_example/lib/models/metier.dart
class Metier {
  final int id;
  final String nom;
  final String description;
  final int salaireMin;
  final int salaireMoyen;
  final int salaireMax;

  const Metier({
    required this.id,
    required this.nom,
    required this.description,
    required this.salaireMin,
    required this.salaireMoyen,
    required this.salaireMax,
  });

  factory Metier.fromJson(Map<String, dynamic> json) {
    return Metier(
      id: json['id'] as int,
      nom: json['nom'] as String,
      description: json['description'] as String,
      salaireMin: json['salaire_min'] as int,
      salaireMoyen: json['salaire_moyen'] as int,
      salaireMax: json['salaire_max'] as int,
    );
  }
}
